src/coreclr/build-runtime.cmd

```diff
@@ -355,7 +357,12 @@ if %__BuildNative% EQU 1 (
         set __ExtraCmakeArgs="-DCMAKE_BUILD_TYPE=!__BuildType!"
     )
 
-    set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLR_CMAKE_TARGET_ARCH=%__TargetArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%" %__CMakeArgs%
+    :: When the host runs on an unknown rid, it falls back to the output rid
+    :: Strip the architecture
+    for /f "delims=-" %%i in ("%__OutputRid%") do set __FallbackOS=%%i
+    if "%__FallbackOS%" == "" set __FallbackOS=win10
+
+    set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLR_CMAKE_TARGET_ARCH=%__TargetArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%" "-DCLI_CMAKE_FALLBACK_RID=%__FallbackOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%" %__CMakeArgs%
```

Comment:
This what shows up in the Windows build log:

> Calling "D:\a\_work\1\s\eng\native\gen-buildsys.cmd" "D:\a\_work\1\s\src\coreclr" "D:\a\_work\1\s\artifacts\obj\coreclr\windows.x86.Checked" vs2022 x86 windows "-DCMAKE_BUILD_TYPE=Checked" "-DCLR_CMAKE_TARGET_ARCH=x86" "-DCLR_CMAKE_TARGET_OS=windows" "-DCLI_CMAKE_FALLBACK_RID=" "-DCLR_CMAKE_PGO_INSTRUMENT=0" "-DCLR_CMAKE_OPTDATA_PATH=" "-DCLR_CMAKE_PGO_OPTIMIZE=0" 

note: `-DCLI_CMAKE_FALLBACK_RID=` is empty.

I have no clue what I'm doing wrong.

Reply:
And the previous line is `The system cannot find the drive specified.`

`for /f` has two meaning, it can treat the argument as file path (and reads its text) or string text, when argument is quoted. Somehow, it doesn't like `for` inside `if` although delayed expansion is enabled. But I'm not a batch expert either. 😅

If we move line 360-363 before the parent `if` block (which starts line 332), then it seems to work as expected.

Reply:
Thanks! Let's give that a run.